### PR TITLE
Add mailing list modal to landing page (#269)

### LINKED ISSUE #23: Tests are broken
Seems that some files are missing for the test script to work correctly:

```shell
[william@william-pc coding-coach]$ yarn test
yarn run v1.9.4
$ node scripts/custom-config test --env=jsdom
/home/william/Dev/Self/WNemencha/coding-coach/node_modules/proxyquire/lib/proxyquire.js:161
    throw err
    ^

Error: Cannot find module '../utils/createJestConfig' from '/home/william/Dev/Self/WNemencha/coding-coach/node_modules/react-scripts/scripts'
    at Function.module.exports [as sync] (/home/william/Dev/Self/WNemencha/coding-coach/node_modules/resolve/lib/sync.js:43:15)
    at Proxyquire._resolveModule (/home/william/Dev/Self/WNemencha/coding-coach/node_modules/proxyquire/lib/proxyquire.js:137:20)
    at Proxyquire.<anonymous> (/home/william/Dev/Self/WNemencha/coding-coach/node_modules/proxyquire/lib/proxyquire.js:205:35)
    at Array.reduce (<anonymous>)
    at Proxyquire._withoutCache (/home/william/Dev/Self/WNemencha/coding-coach/node_modules/proxyquire/lib/proxyquire.js:204:6)
    at Proxyquire.load (/home/william/Dev/Self/WNemencha/coding-coach/node_modules/proxyquire/lib/proxyquire.js:129:15)
    at Object.<anonymous> (/home/william/Dev/Self/WNemencha/coding-coach/scripts/custom-config.js:39:5)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### LINKED ISSUE #34: Account Settings
**Acceptance Criteria:**

A user can update the following required fields:
- Name
- Email address
- Job title
- Languages the user speaks

A user can update the following optional fields:
- Social media accounts
- Company
- Skills to mentor in (depends on account type)
- Skills to be mentored in depends on account type)

Users will be able to set their account type to be mentor, mentee, or both.

A mentor can update the following fields:
- Skills they are able to mentor in
- Proficiency of skills they are able to mentor in

A mentee can update the following fields:
- Skills they are looking to improve

All users are able to set themselves as a mentor.

### LINKED ISSUE #42: Delete Account
**Acceptance Criteria:**
- Users can delete their accounts.
- This will remove all data from the database.
- All current mentorships will be notified.

### PR DESCRIPTION
**Pull request by alizand1992**
* public/index.html
  Add the script to load the Mailchimp library

* src/pages/home/Home.js
  Add the MailingModal to the home page

* src/pages/home/components/MailingModal.js
  Create the MailingModal class component to open up the modal.

# Details
## Description
* I added the first script tag to the index.html. This loads up the form into the site. 
* In order to call the modal that contains the form, I have created a new component called MailingModal within the home component directory.
  * Tests were failing on window.dojoRequire (not a function) so I added an undefined check. it seems that the dojoRequire is getting loaded up to the window scope via the script tag in the index.html

## Steps to QA
- [ ] Start the application
- [ ] After loading the landing page, the modal will appear within a few seconds
- [ ] Once the user is subscribed, a thank you modal will appear for a few seconds and then disappears. 
- [ ] The next time the page loads the modal should not appear.

## Screenshots

coding-coach-mailing-list-modal

## Issue
#269

### CODE REVIEW

**Comment by emmawedekind** [comment]
This is great, I just wonder why the modal is so big... our image isn't showing up. I just updated MailChimp. Could you try this? 

```
<script type="text/javascript" src="//downloads.mailchimp.com/js/signup-forms/popup/unique-methods/embed.js" data-dojo-config="usePlainJson: true, isDebug: false"></script><script type="text/javascript">window.dojoRequire(["mojo/signup-forms/Loader"], function(L) { L.start({"baseUrl":"mc.us19.list-manage.com","uuid":"98c69835a5abaf1759e7618ec","lid":"3f9f14ba16","uniqueMethods":true}) })</script>
```

**Comment by alizand1992** [comment]
@emmawedekind 

It could have possibly been something on MailChimp's side. I just cleared my cookies and reloaded the page and got the following:

coding-coach-mailing-list-modal-2

I've recently started with React so I could be totally mistaken here.

**Comment by emmawedekind** [comment]
Wonderful!! Looks awesome :)

**Comment by emmawedekind** [comment]
@crysfel Whenever you're back, take a look at this! Let's push it as soon as possible :)

**Comment by trent-boyd** [comment]
I think if we're going to let MailChimp handle all the modal creation, form styling, and state management, we should probably just get rid of the React portion of it and put the entirety of the script into our `index.html`. It would be easier to keep track of, and Jest wouldn't ever see it.

Does anybody else have any thoughts?

**Comment by emmawedekind** [comment]
> I think if we're going to let MailChimp handle all the modal creation, form styling, and state management, we should probably just get rid of the React portion of it and put the entirety of the script into our `index.html`. It would be easier to keep track of, and Jest wouldn't ever see it.
> 
> Does anybody else have any thoughts?

I totally agree. Can you please implement @alizand1992 :)

File: src/pages/home/components/MailingModal.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          uuid: '98c69835a5abaf1759e7618ec',
+          lid: '3f9f14ba16',
+          uniqueMethods: true,
+        });
+      });
+    }
+  }
+
+  render() {
+    return <div />;
```
Comment (trent-boyd) [inline src/pages/home/components/MailingModal.js]: FWIW in the future, you _can_ return `null` from a React render method. :)

**Comment by emmawedekind** [comment]
Wonderful! I will pull & check it out :)

**Comment by emmawedekind** [comment]
Closes #269

**Comment by emmawedekind** [comment]
This looks totally great!!! Thank you!

**Comment by emmawedekind** [comment]
@crysfel If you could take a look - I think we're ready to merge!

**Comment by tureey** [comment]
@crysfel he's taking really seriously the vacations hahaha

**Comment by crysfel** [comment]
hehehe sorry for the late response, yeap I was completely off the grid xD

**Comment by alizand1992** [comment]
Awesome, thanks everyone for your help. I'm happy to be contributing.

**Pull request merged**